### PR TITLE
CodeEditor: Improved styles when the code editor is loading

### LINKED ISSUE #80861: Query input: no space around "Loading..." placeholder
### What happened?

I often see this for a few milliseconds and it annoys me but then I forget and get back to work.

image

However, today I'm working on a very bad internet connection and I had to stare at that "Loading..." placeholder for a while.

### What did you expect to happen?

I would expect the "Loading..." placeholder to be styled consistently with the rest of the interface.

### Did this work before?

I don't think so.

### How do we reproduce it?

1. Open "Explore" on a slow connection (simulate from Chrome)
2. Try not to cry while staring at the "Loading..." without proper padding.

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: Grafana v10.2.3 (1e84fede54)
OS: Mac
Browser: Chrome, Arc.

### Grafana platform?

I use Grafana Cloud

### Datasource(s)?

Prometheus

### PR DESCRIPTION
**Pull request by NWRichmond**
## What is this change?

Style & messaging updates to the Monaco code editor, in the following places:
- `grafana-ui`'s `ReactMonacoEditorLazy` component (exported as `ReactMonacoEditor`), which is currently used by the Prometheus and Loki data sources directly. It's also used in `grafana-ui`'s `CodeEditor` component, which is used in the Azure Monitor, CloudWatch, InfluxDB, Parca, and Tempo data sources, as well as the Panel plugin.
- the Prometheus data source

## Why do we need this feature?

As described in #80861, the loading UI for the Monaco code editor used by the Prometheus data source (and other data sources) has styling that's inconsistent with the rest of the UI.

## Who is this feature for?

Users who rely on code editor UI in various data sources.

## Which issue(s) does this PR fix?

Fixes #80861

## Special notes for your reviewer

Please check that:

- [x] It works as expected from a user's perspective.
  - [x] Prometheus data source
  - [x] Loki data source
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Monaco/ReactMonacoEditorLazy.tsx [hunk 1]
```diff
@@ -11,13 +16,14 @@ import type { ReactMonacoEditorProps } from './types';
...
-export const ReactMonacoEditorLazy = (props: ReactMonacoEditorProps) => {
+const MonacoEditorLazy = (props: ReactMonacoEditorProps) => {
+  const styles = useStyles2(getStyles);
   const { loading, error, dependency } = useAsyncDependency(
     import(/* webpackChunkName: "react-monaco-editor" */ './ReactMonacoEditor')
   );
 
   if (loading) {
-    return <LoadingPlaceholder text={''} />;
+    return <LoadingPlaceholder text={''} className={styles.container} />;
```
Comment (matyax) [inline packages/grafana-ui/src/components/Monaco/ReactMonacoEditorLazy.tsx:26]: What do you think about adding some text like "Loading editor" along with the spinner. I feel that it looks kind of lonely without it.

imagen

**Review by matyax — APPROVED** [review]
Thank you @NWRichmond ! Approving logs changes.

File: packages/grafana-ui/src/components/Monaco/ReactMonacoEditorLazy.tsx (hunk 1 above)
Comment (NWRichmond) [reply packages/grafana-ui/src/components/Monaco/ReactMonacoEditorLazy.tsx]: Yes, good suggestion!

https://github.com/grafana/grafana/assets/5732000/a47e6363-2d52-451c-8e16-e6eaec6b36ec

Implemented in [009e08c](https://github.com/grafana/grafana/pull/88102/commits/009e08c946988b5d023def0907aa392181ae3f0e). I'm very open to this and other ideas about the placeholder contents.

**Comment by gtk-grafana** [comment]
You might need to tweak some e2e tests, I seem to remember a few targeting the loading state, but maybe that's been fixed 🤷

File: packages/grafana-ui/src/components/Monaco/ReactMonacoEditorLazy.tsx (hunk 1 above)
Comment (itsmylife) [reply packages/grafana-ui/src/components/Monaco/ReactMonacoEditorLazy.tsx]: Just a nit: Wouldn't it be nice to have loading spinner before the text?

Comment (NWRichmond) [reply packages/grafana-ui/src/components/Monaco/ReactMonacoEditorLazy.tsx]: @itsmylife perhaps, but that is governed by the [`<LoadingPlaceholder />`](https://developers.grafana.com/ui/latest/index.html?path=/story/general-loadingplaceholder--basic).

**Comment by NWRichmond** [comment]
Judging by the passing integration tests, I think we're good to go. Merging :)

**Pull request merged**